### PR TITLE
Added Segment.io opt-int logic

### PR DESCRIPTION
**Pull request by lucferbux**
- [X] The commits are squashed in a cohesive manner and have meaningful messages.
- [X] For commits that came from upstream, `[UPSTREAM]` has been prepended to the commit message
- [X] JIRA link(s): https://issues.redhat.com/browse/RHODS-871
- [X] The Jira story is acked
- [X] An entry has been added to the latest build document in [Build Announcements Folder](https://drive.google.com/drive/folders/1sgkK1WZgGo9CXsLizNe0GbAzVKuSKrZL).
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious)

## Testing instructions
1. Deploy this PR with the [ODH-Dashboard PR](https://github.com/red-hat-data-services/odh-dashboard/pull/178).
2. Ensure both `segment-key-secret` secret and `rhods-segment-key-config` configmap are deployed in the apps Namespace.
3. Check that the `rhods-segment-key-config` is set to **true**.
4. Open odh dashboard and inspect network, you should find a network call called `/segment-key` with a *write key* enabled.
odh_dashboard_true

5. Additionally segment should be tracking the interaction.
6. Now change `rhods-segment-key-config` to **false**.
7. Reload odh dashboard and now `/segment-key` should be empty.
odh_dashboard_false

### CODE REVIEW

**Comment by lucferbux** [comment]
@anishasthana @gmfrasca one thing to review is that the secret is pointing to the actual writing key of our segment deployment.

File: backend/src/routes/api/segment-key/segmentKeyUtils.ts [hunk 1]
```diff
@@ -3,14 +3,20 @@ import { KubeFastifyInstance, ODHSegmentKey } from '../../../types';
...
   const namespace = fastify.kube.namespace;
+  let segmentKeyEnabled = true;
   try {
+    const resEnabled = await coreV1Api.readNamespacedConfigMap(
+      'rhods-segment-key-enabled',
+      namespace,
+    );
+    segmentKeyEnabled = resEnabled?.body.data?.segmentKeyEnabled === 'true';
     const res = await coreV1Api.readNamespacedSecret('rhods-segment-key', namespace);
     const decodedSegmentKey = Buffer.from(res.body.data.segmentKey, 'base64').toString();
```
Comment (jeff-phillips-18) [inline backend/src/routes/api/segment-key/segmentKeyUtils.ts:14]: Is there any point in getting the segment key if it is not enabled?

Comment (lucferbux) [reply backend/src/routes/api/segment-key/segmentKeyUtils.ts]: Yeah, you are completly right, updated!

**Review by dlabaj — APPROVED** [review]
lgtm

**Review by jeff-phillips-18 — APPROVED** [review]

**Comment by lucferbux** [comment]
mmm i'm having the same integration issue as @dlabaj had in his PR, not sure if remove the integration check here too, as it could generate a merge conflict, thoughts? @jeff-phillips-18 @DaoDaoNoCode

**Comment by lucferbux** [comment]
> @lucferbux I'd do it for upstream as well. Can fix any conflicts on the downstream merge.

Great!

**Review by anishasthana — APPROVED** [review]
We tested with http://quay.io/lferrnan/rhods-operator-live-catalog:1.8.3-segment-final. Changes work!

**Review by anishasthana — APPROVED** [review]

**Pull request merged**